pulsar-jms/src/main/java/com/datastax/oss/pulsar/jms/PulsarConnectionFactory.java

```diff
@@ -1037,6 +1042,10 @@ public void close() {
     }
   }
 
+  Map<Object, Producer<byte[]>> getProducers() {
+    return isUseTemporaryProducers() ? tempProducers : producers;
```

Comment:
the decision of using temporary producers is per factory, you don't need to hashmaps